Tasks/AzureFileCopyV6/AzureUtilityAz1.0.ps1

```diff
@@ -2,22 +2,46 @@
 
 . "$PSScriptRoot/AzureUtilityRest.ps1"
 
-function Get-AzureStorageAccountResourceGroupName
-{
+function Get-AzureStorageAccountResourceGroupName {
     param([string]$storageAccountName)
 
-    $ARMStorageAccountResourceType =  "Microsoft.Storage/storageAccounts"
-    if (-not [string]::IsNullOrEmpty($storageAccountName))
-    {
-        Write-Verbose "[Azure Call]Getting resource details for azure storage account resource: $storageAccountName with resource type: $ARMStorageAccountResourceType"
-        $azureStorageAccountResourceDetails = Az.Storage\Get-AzStorageAccount -ErrorAction Stop | Where-Object { $_.StorageAccountName -eq $storageAccountName } 
-          
-        Write-Verbose "[Azure Call]Retrieved resource details successfully for azure storage account resource: $storageAccountName with resource type: $ARMStorageAccountResourceType"
+    $ARMStorageAccountResourceType = "Microsoft.Storage/storageAccounts"
+
+    if (-not [string]::IsNullOrEmpty($storageAccountName)) {
+        Write-Verbose "[Azure Call] Getting resource details for Azure storage account resource: $storageAccountName with resource type: $ARMStorageAccountResourceType"
 
+        $maxRetries = 3
+        $retryDelay = 30
+
+        for ($retryCnt = 0; $retryCnt -le $maxRetries; $retryCnt++) { 
+            try {
+                # Attempt to get the Azure Storage Account details
+                $azureStorageAccountResourceDetails = Az.Storage\Get-AzStorageAccount -ErrorAction Stop |
+                    Where-Object { $_.StorageAccountName -eq $storageAccountName }
+                # If successful, exit the loop
+                break
+            }
+            catch {
+                $errorMessage = $_.Exception.Message                 
+                # Retry logic for HTTP 429 (Too Many Requests)
+                if ($_.Exception.Response.StatusCode -eq 429) {
+                    Write-Verbose "Exception Message: $($_.Exception.Response.Message)"
+                    Write-Verbose "Exception Response StatusCode: $($_.Exception.Response.StatusCode)"
+                    # Wait before retrying
+                    Start-Sleep -Seconds $retryDelay
+                    continue
+                }
+                else {
+                    # For other errors, display the message and exit the loop
+                    Write-Verbose "[Error]: $errorMessage"
+                    break
+                }
+            }
+        }
+        Write-Verbose "[Azure Call] Retrieved resource details successfully for Azure storage account resource: $storageAccountName with resource type: $ARMStorageAccountResourceType"
```

Comment:
Let's write this before Break on 22 line since we get here even if we failed.

Reply:
Alse, I'd suggest putting the error message into the `task.json` messages and calling `Get-VstsLocString -Key` for localization support.

Reply:
Sure